### ISSUE TITLE
Japanese dictionary deconjugation and capturing clipboard text

### ISSUE THREAD

**Issue by shiki-tm** [Jimex/DictTango-Android#24]
I don't know anything about coding or about the mdx/mdd formats so I apologize if this is a stupid question.

I was wondering if deconjugation of Japanese dictionaries is supported by your app? I have a whole collection of Japanese dictionaries but if I ever search for a word  that's conjugated, I get 0 search results back. I just wanted to know if that's a limitation of my dictionaries, the app or of the dictionary format. 

Also, how do I get the "capture clipboard text" setting working? I'm not sure what the word lookup mode is. Is it possible to capture clipboard text while looking up a word in another app through the context menu (copy)?
Since dict tango doesn't show up in the context menu depending on the app. Is there any way to fix that? 

Thank you for continuing to update this app! 
I look forward to your reply.

**Comment by Jimex** [comment]
> I was wondering if deconjugation of Japanese dictionaries is supported by your app? I have a whole collection of Japanese dictionaries but if I ever search for a word that's conjugated, I get 0 search results back. I just wanted to know if that's a limitation of my dictionaries, the app or of the dictionary format.

Can you upload the mdx file and the sample word for me to investigate?

> Also, how do I get the "capture clipboard text" setting working? I'm not sure what the word lookup mode is. Is it possible to capture clipboard text while looking up a word in another app through the context menu (copy)?
Since dict tango doesn't show up in the context menu depending on the app. Is there any way to fix that?

The "Clipboard" function has been disabled in Android 9 and newer versions. To quickly look up words using DictTango, follow these steps:

1. Navigate to settings and enable "Capture Clipboard text."
2. Return to the main screen and tap "Foreground Service."
3. Activate both "Enable foreground service" and "Display floating button" on the foreground service screen.
4. After copying text from any app, tap the DictTango floating button, then select "Text that you copied" from the pop-up window.

**Comment by cristian428** [comment]
Thanks for explaining the capture clipboard text feature! 

Jitendex dictionary can be downloaded here:
https://jitendex.org/pages/downloads.html

Any conjugated word is fine as an example such as "まくってる" from 捲る.

**Comment by Jimex** [comment]
I'm not quite following your point. Can you provide a screen recording for a demonstration?

![image](https://github.com/Jimex/DictTango-Android/assets/3145858/4d27bf89-f1bd-4494-91af-720cbf749453)

**Comment by cristian428** [comment]
https://github.com/Jimex/DictTango-Android/assets/36088384/e4a4977a-7c6f-41fb-8d3f-e68c47f95a4d

With the occasional exception of 1 dictionary that only is used for redirecting, none of the dictionaries show anything. I'm in the correct dictionary group, so I don't think I'm doing something wrong. I'm not sure if this is expected behavior from the app or not. 



I just thought of a good, unrelated idea: 
Can you add a button to the notification bar to turn on the Foreground service ? 
A button like in this screenshot:
![Screenshot_20240511_124939_DX](https://github.com/Jimex/DictTango-Android/assets/36088384/0085e353-432b-4f30-a4e6-8931bc4a469e)

Thank you for taking the time to talk to me.

**Comment by Jimex** [comment]
> but if I ever search for a word that's conjugated, I get 0 search results back.

let's focus on this first, you mentioned that you get 0 search results back, but from the screen recording I can see that there is a result

**Comment by cristian428** [comment]
That result is not useful, and doesn't show up for lots of words. Just happened to show up for that word. I really think there's a problem or a lack of deconjugation. 

Also, I can't select dict tango in my context menu. That's another problem I've noticed. Sometimes the button does appear but sometimes it doesn't. Depends on the app. Usually if it's the "white" context menu, it doesn't appear but if it's the "black"  context menu, it does. 

![Screenshot_20240511_131008_DX](https://github.com/Jimex/DictTango-Android/assets/36088384/1f22b73c-27ac-4905-8bb4-219966ed2a26)
![Screenshot_20240511_131410_Kiwi Browser](https://github.com/Jimex/DictTango-Android/assets/36088384/53dc016b-2233-4ef4-ac3b-90ff64626ef9)

**Comment by Jimex** [comment]
When you tap on a word in the list, it will display its content. To view the word list, enter a word in the search box at the top.

**Comment by cristian428** [comment]
Yes but if the verb or adjective is conjugated, like most of the time, then sometimes you can't know what's the dictionary form of a word and it won't show up in the search results..

**Comment by KonstantinDjairo** [comment]
it's not trivial to implement a deinflector, but it's possible
but that would affect how dictTango treats input in general, so if someone is willing to implement a japanese deinflector for DictTango, i suggest adding an option to enable it, rather than having it enabled by default

an example of working deinflector is:
https://git.ajattix.org/hashirama/rdricpp
